### PR TITLE
Patches for custom domain

### PR DESCRIPTION
**Pull request by philschanely**
## Description

Apologies in advance for this large PR. It started as a small set of patches I needed to make for my work on the redesigned Custom Domain setup and bubbled a little bit larger than I expected. This includes the following:

- Improves default spacing around text in `.sage-type` containers, particularly spacing above and below headings and below paragraphs. 🍮 
- Small tweaks to spacing and text sizing in the Banner object 🎏 
- Adds a white background to cards 😳 
- Patches an outdated BEM class on alert object (requires a patch in Products I have staged and ready) 🩹 
- Adds a text coloring utility class for easily adjusting colors (esp for icons!) 🌈 
- Adds a decorator element for within Nav items
- Adds customizaton options for hyperlinks

### Screenshots

|  Thing  |  Thing rendered  |
|--------|--------|
| Type spacing | Screen Shot 2020-07-23 at 8 22 51 AM |
| Text coloring | Screen Shot 2020-07-23 at 8 21 43 AM |
| Nav link decorator | Screen Shot 2020-07-23 at 8 17 49 AM |
| Link modifiers | Screen Shot 2020-07-23 at 8 18 10 AM |

### CODE REVIEW

File: app/views/examples/elements/button/_preview.html.erb [hunk 1]
```diff
@@ -1,678 +1,675 @@
...
-    is_disabled: false,
-    is_link_btn: true,
-    link_url: "",
-    size: "tiny",
-    text: "Name Of Action"
-  %>
-
-</div>
\ No newline at end of file
+<h3 class="t-sage-heading-6">Primary</h3>
```
Comment (philschanely) [inline app/views/examples/elements/button/_preview.html.erb:1]: This is just pulling out the wrapping div and removing some of the `br` tags

File: app/views/examples/elements/_element.html.erb [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 <div class="docs-container">
   <h2 id="<%= @title %>-example-preview" class="t-sage-heading-2 example__title">Preview</h2>
-  <div class="example__preview">
+  <div class="example__preview sage-type">
```
Comment (philschanely) [inline app/views/examples/elements/_element.html.erb:15]: Several like this are just adding the `sage-type` class on examples for better spacing.

File: app/views/examples/elements/label/_preview.html.erb [hunk 3]
```diff
@@ -15,6 +15,7 @@ type: "danger",
 is_subtle: false
 %>
 <br/>
+<br/>
```
Comment (philschanely) [inline app/views/examples/elements/label/_preview.html.erb:18]: I auditing how often we use `br` and tried to remove unnecessary instances.

File: app/views/examples/objects/alert/_markup.html.erb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="sage-alert<%= color != "" ? " sage-alert--#{color}" : "" %>">
   <i class="sage-alert__icon <%= sage_icon_name %>" aria-hidden="true"></i>
-  <div class="sage-alert-copy">
+  <div class="sage-alert__copy">
```
Comment (philschanely) [inline app/views/examples/objects/alert/_markup.html.erb:3]: Proper BEM implementation. Staged corresponding fixes in `products`.

File: app/views/pages/typography.html.erb [hunk 5]
```diff
@@ -152,50 +152,87 @@
     As an example, see here how the "Heading 3" type spec can be applied to an
     <code>h3</code> element:
   </p>
-</div>
 
-<div class="docs-container">
-  <h3 class="t-sage-heading-3">Lorem ipsum dolor sit</h3>
-  <div class="sage-code-snippet">
-    <pre class="prettyprint"><code>&lt;h3 class="t-sage-heading-3"&gt;Lorem ipsum dolor sit&lt;/h3&gt;</code></pre>
+  <div class="docs-container">
```
Comment (philschanely) [inline app/views/pages/typography.html.erb:156]: Here I was simplifying how often we break in and out of a `sage-type` container and just use one for the whole block instead.

File: app/views/pages/typography.html.erb [hunk 6]
```diff
@@ -152,50 +152,87 @@
...
+  &lt;/div&gt;</code></pre>
+    </div>
   </div>
-  <div class="sage-code-snippet">
-    <pre class="prettyprint"><code>&lt;div class="sage-type"&gt;
-  &lt;h3&gt;User-generated content&lt;/h3&gt;
-  &lt;p&gt;See how content inside this &lt;code&gt;sage-type&lt;/code&gt; container automatically recieves corresponding type specs?&lt;/p&gt;
-&lt;/div&gt;</code></pre>
+
+  <h2>Coloring Type</h2>
```
Comment (philschanely) [inline app/views/pages/typography.html.erb:199]: New bit on text coloring classes.

File: lib/sage-frontend/stylesheets/system/core/_typography.scss [hunk 7]
```diff
@@ -224,3 +264,19 @@ hr {
   border: 0;
   border-top: sage-border();
 }
+
```
Comment (philschanely) [inline lib/sage-frontend/stylesheets/system/core/_typography.scss:267]: New color classes setup

**Review by pixelflips — APPROVED** [review]
👍  - Nice work and thanks for the additional clean up! Those new link styles will come in handy in an upcoming project I have as well.

**Pull request merged**